### PR TITLE
Fixes 238, by requiring vol_idx input with 4D images.

### PR DESCRIPTION
**Pull request by arokem**
See #238

### CODE REVIEW

**Comment by skoudoro** [comment]
Great, Thank you for that! Waiting for Travis...

**Comment by skoudoro** [comment]
Great! Thank you for this fix! As you can see [here](https://travis-ci.org/nipy/dipy/jobs/527946477#L6347) it seems you need to update some workflow test. After that, I will go ahead and merge it

**Comment by arokem** [comment]
Yeah. I saw that. Will fix!

**Comment by arokem** [comment]
OK. I think that this is ready to be reviewed. I believe that remaining CI failures are unrelated.

**Comment by arokem** [comment]
Is this the doc you meant?

**Comment by skoudoro** [comment]
yes, perfect 👍

**Comment by skoudoro** [comment]
Hi @arokem, Do you understand the Travis error? I am surprised since everything is good at #1775

**Comment by arokem** [comment]
Not really,  and I don't understand why it passed there, but not here... 

But maybe it has to do with this line: https://github.com/nipy/dipy/blob/master/.travis.yml#L64 ?

Why are we pointing to python2 here?

**Comment by skoudoro** [comment]
Ok, I see, good catch. It was mainly because of VTK. But we do not need that anymore since we install VTK via Pypi.

**Comment by arokem** [comment]
I can try fixing that here.

File: .travis.yml [hunk 1]
```diff
@@ -59,12 +59,8 @@ matrix:
       env:
         - COVERAGE=1
         - VTK=1
-        - VTK_VER="python-vtk"
-        - LIBGL_ALWAYS_INDIRECT=y
-        - VENV_ARGS="--system-site-packages --python=/usr/bin/python2.7"
         - TEST_WITH_XVFB=true
-        - DEPENDS="$DEPENDS scikit_learn fury"
-        - MESA_GL_VERSION_OVERRIDE=3.3
```
Comment (skoudoro) [inline .travis.yml]: sorry, you have to keep: `MESA_GL_VERSION_OVERRIDE=3.3`,  `LIBGL_ALWAYS_INDIRECT=y`, `DEPENDS="$DEPENDS scikit_learn fury"` but remove [this line](https://github.com/nipy/dipy/blob/master/.travis.yml#L117)

**Comment by skoudoro** [comment]
Thank you @arokem, merging

**Pull request merged**